src/Response.php

```diff
@@ -277,7 +287,7 @@ public function setServiceMap($serviceMap)
      *
      * @return Smd|null
      */
-    public function getServiceMap()
+    public function getServiceMap() : ?Smd
```

Comment:
This is a bit confusing - in theory `->serviceMap` can be `null`, but only when it is not initialised.
We would need to decide how we are doing it. I think for that case I would prefer drop nullable type here.
With PHP 7.4 if we add property type declaration (`protected Smd $serviceMap;`) we get error when we try get value of uninitialised property.

Reply:
As in the current implementation the `$serviceMap` property is optional, nullable looks fine to me. Removing the nullable will break `Response\Http::sendHeaders()`.
This seems to be an issue that needs to addressed with update to PHP 7.4.

Reply:
If this is optional I think setter should allow nullable values. Thoughts?

Reply:
Agreed.